src/main/java/org/folio/auth/authtokenmodule/impl/ModulePermissionsSource.java

```diff
@@ -155,27 +151,23 @@ private void endRequest(HttpClientRequest req, String requestToken,
     String tenant, String okapiUrl, String requestToken, String requestId) {
 
     final String key = tenant + "_" + permissions.encodePrettily();
-    Future<JsonArray> future = Future.future();
     CacheEntry<JsonArray> entry = expandPermissionsMap.get(key);
     if (entry != null && entry.getAge() < expandPermissionsTimeout) {
-      future.complete(entry.getEntry());
-      return future;
+      return Future.succeededFuture(entry.getEntry());
     }
-    future = expandPermissions(permissions, tenant, okapiUrl, requestToken, requestId);
+    Future<JsonArray> future = expandPermissions(permissions, tenant, okapiUrl, requestToken, requestId);
     return future.compose(res -> {
-      Future<JsonArray> future2 = Future.future();
       expandPermissionsMap.put(key, new CacheEntry<>(res));
-      future2.complete(res);
-      return future2;
+      return Future.succeededFuture(res);
     });
   }
 
   private Future<JsonArray> expandPermissions(JsonArray permissions, String tenant,
     String okapiUrl, String requestToken, String requestId) {
-    Future<JsonArray> future = Future.future();
+    Promise<JsonArray> promise = Promise.promise();
     if (permissions.isEmpty()) {
-      future.complete(new JsonArray());
-      return future;
+      promise.complete(new JsonArray());
+      return promise.future();
     }
```

Comment:
Minor suggestion: the **promise** declaration can be moved down to save an extra line.
```
    if (permissions.isEmpty()) {
      return Future.succeededFuture(new JsonArray());
    }
    Promise<JsonArray> promise = Promise.promise();
```

Reply:
Yep